transform-plugins/src/test/java/co/cask/hydrator/plugin/DecoderTest.java

```diff
@@ -0,0 +1,196 @@
+/*
+ * Copyright © 2015 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package co.cask.hydrator.plugin;
+
+import co.cask.cdap.api.data.format.StructuredRecord;
+import co.cask.cdap.api.data.schema.Schema;
+import co.cask.cdap.etl.api.Transform;
+import org.apache.commons.codec.binary.Base32;
+import org.apache.commons.codec.binary.Base64;
+import org.apache.commons.codec.binary.Hex;
+import org.junit.Assert;
+import org.junit.Test;
+
+/**
+ * Tests {@link Decoder}
+ */
+public class DecoderTest {
+  private static final Schema INPUT = Schema.recordOf("input",
+                                                      Schema.Field.of("a", Schema.of(Schema.Type.STRING)),
+                                                      Schema.Field.of("b", Schema.of(Schema.Type.STRING)),
+                                                      Schema.Field.of("c", Schema.of(Schema.Type.STRING)),
+                                                      Schema.Field.of("d", Schema.of(Schema.Type.STRING)),
+                                                      Schema.Field.of("e", Schema.of(Schema.Type.STRING)));
+
+  private static final Schema OUTPUT = Schema.recordOf("output",
+                                                       Schema.Field.of("a", Schema.of(Schema.Type.BYTES)),
+                                                       Schema.Field.of("b", Schema.of(Schema.Type.STRING)));
+
+  private static final Schema OUTPUTSTR = Schema.recordOf("outputstr",
+                                                       Schema.Field.of("a", Schema.of(Schema.Type.STRING)),
+                                                       Schema.Field.of("b", Schema.of(Schema.Type.STRING)));
+
+  @Test
+  public void testBase64Decoder() throws Exception {
+    String test = "This is a test for testing base64 decoding";
+    Transform<StructuredRecord, StructuredRecord> encoder =
+      new Encoder(new Encoder.Config("a:BASE64", OUTPUT.toString()));
+    encoder.initialize(null);
```

Comment:
Instead of passing null, pass in `MockTransformContext`, in case the decoder transform initialize method does something with the context or if the superclass `Transform` does something with the context passed into the initialize method, right?